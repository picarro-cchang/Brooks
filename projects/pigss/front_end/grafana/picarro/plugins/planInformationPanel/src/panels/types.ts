import { any } from "prop-types";

export interface Plan {
  max_steps: number;
  panel_to_show: number;
  current_step: number;
  focus: PlanFocus;
  last_step: number;
  steps: { [key: string]: PlanStep };
  num_plan_files: number;
  plan_files: { [key: string]: string };
  plan_filename: string;
  bank_names: {
    [key: number]: {
      name: string;
      channels: { [key: number]: string };
    };
  };
}

export interface PlanFocus {
  row: number;
  column: number;
}

export interface BankConfig {
  chan_mask: number;
  clean: number;
}

export interface PlanStep {
  banks: { [key: number]: BankConfig };
  reference: number;
  duration: number;
}
