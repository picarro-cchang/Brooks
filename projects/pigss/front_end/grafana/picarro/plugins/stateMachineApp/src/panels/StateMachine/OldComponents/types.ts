// import { any } from "prop-types";

// export interface ButtonInfo {
//   caption: string;
//   className: string;
//   response: string;
// }

// export interface CommandPanelOptions {
//   uistatus: {
//     [key: string]: string;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;
//   layoutSwitch: () => {}
// }

// export interface ModalInfo {
//   show: boolean;
//   html: string;
//   num_buttons: number;
//   buttons: { [key: string]: ButtonInfo };
// }

// export interface Plan {
//   max_steps: number;
//   panel_to_show: number;
//   current_step: number;
//   focus: PlanFocus;
//   last_step: number;
//   steps: { [key: string]: PlanStep };
//   num_plan_files: number;
//   plan_files: { [key: string]: string };
//   plan_filename: string;
//   bank_names: {
//     [key: number]: {
//       name: string;
//       channels: { [key: number]: string };
//     };
//   };
// }

// export interface PlanFocus {
//   row: number;
//   column: number;
// }

// export interface PlanLoadPanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   updatePanel: (x: number) => void;
//   loadPlan: (x: string) => void;
//   deleteFile: (x: string) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
//   cancelLoadPlan: () => void;
// }

// export interface PlanLoadPanelEditOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   getPlanFromFileName: (s: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export interface PlanPanelOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   fileName: string;
//   loadFile: () => void;
//   savePlan: () => void;
//   updatePanel: (x: number) => void;
// }

// export interface PlanPanelLayoutOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   updatePanel: (x: number) => void;            
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export enum PlanPanelTypes {
//   NONE = 0,
//   PLAN = 1,
//   LOAD = 2,
//   EDIT = 3,
//   PREVIEW = 4
// }

// export interface PlanSavePanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   planSaved: (f: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   fileNames: {[key: string]: string}
// }

// export interface BankConfig {
//   chan_mask: number;
//   clean: number;
// }

// export interface PlanStep {
//   banks: { [key: number]: BankConfig };
//   reference: number;
//   duration: number;
// }

// export interface EditPanelOptions {
//   uistatus: {
//     [key: string]: any;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;

// }
