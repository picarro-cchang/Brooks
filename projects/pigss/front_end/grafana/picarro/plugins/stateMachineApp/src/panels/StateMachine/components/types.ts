export interface PlanFocus {
    row: number;
    column: number;
}

export interface BankConfig {
    chan_mask: number;
    clean: number;
}

export interface PlanStep {
    banks: { [key: number]: BankConfig };
    reference: number;
    duration: number;
}

export interface Plan {
    max_steps: number;
    panel_to_show: number;
    current_step: number;
    focus: PlanFocus;
    last_step: number;
    steps: { [key: string]: PlanStep };
    num_plan_files: number;
    plan_files: { [key: string]: string };
    plan_filename: string;
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
}

export interface PlanLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    getPlanFileNames: () => void;
    loadedFileName: string;
}

export interface RunLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    loadedFileName: string;
    getPlanFileNames: () => void;
}

export enum PanelTypes {
    NONE = 0,
    LOAD = 1,
    EDIT = 2,
    PREVIEW = 3
}

export enum PlanPanelTypes {
    PLAN = 0,
    LOAD = 1,
    SAVE = 2,
}

export interface BankPanelPlanOptions {
    bank: number;
    uistatus: {
      bank?: { [bankNum: string]: string };
      clean?: { [bankNum: string]: string };
      channel?: { [bankNum: string]: { [channelNum: string]: string } };
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    addChanneltoPlan: (bank: number, channel: number) => void;
}

export interface LoadPanelCommandOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
    planID: String;
  }

  export interface PreviewPanelOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }


  export interface LoadPanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
  }

export interface PlanPanelOptions {
    uistatus: { [key: string]: any };
    plan: Plan;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    bankAddition: { [key: string]: number };
    fileName: string;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    planOverwrite: (s: object, x: String) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    updateSavedFileState: (plan: Plan) => void;
    planID: String;
  }

  export interface PlanSavePanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    planSaved: (f: string, d: object) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    deleteFile: (file: string) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
  }

  export interface EditPanelOptions {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    planID: String;
  
  }
  export interface CommandPanelOptions {
    uistatus: {
      [key: string]: string;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }
