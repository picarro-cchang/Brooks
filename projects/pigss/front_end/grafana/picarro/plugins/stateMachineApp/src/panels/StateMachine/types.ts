import { any } from "prop-types";

export interface ButtonInfo {
  caption: string;
  className: string;
  response: string;
}

export interface CommandPanelOptions {
  uistatus: {
    [key: string]: string;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}

export interface ModalInfo {
  show: boolean;
  html: string;
  num_buttons: number;
  buttons: { [key: string]: ButtonInfo };
}

export interface Plan {
  max_steps: number;
  panel_to_show: number;
  current_step: number;
  focus: PlanFocus;
  last_step: number;
  steps: { [key: string]: PlanStep };
  num_plan_files: number;
  plan_files: { [key: string]: string };
  plan_filename: string;
  bank_names: {
    [key: number]: {
      name: string;
      channels: { [key: number]: string };
    };
  };
}

export interface PlanFocus {
  row: number;
  column: number;
}

export interface PlanLoadPanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface PlanPanelOptions {
  uistatus: { [key: string]: string };
  plan: Plan;
  setFocus: (row: number, column: number) => void;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
  // addChanneltoPlan: (bank: number, channel: number) => void;
  bankAddition: {[key: string]: number}
}

export enum PlanPanelTypes {
  NONE = 0,
  PLAN = 1,
  LOAD = 2,
  SAVE = 3,
  EDIT = 4
}

export interface PlanSavePanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface BankConfig {
  chan_mask: number;
  clean: number;
}

export interface PlanStep {
  banks: { [key: number]: BankConfig };
  reference: number;
  duration: number;
}

export interface EditPanelOptions {
  uistatus: {
    [key: string]: any;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}
