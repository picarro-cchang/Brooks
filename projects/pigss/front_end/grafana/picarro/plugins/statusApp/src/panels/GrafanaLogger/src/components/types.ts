import { GrafanaTheme } from '@grafana/ui';
import { TimeRange } from '@grafana/data';

export interface LogProps {
  level: object[];
  limit: number;
  timeRange: TimeRange;
  data: string[][];
}

export interface LogSectionProps {
  options: LogProps;
  theme: GrafanaTheme;
}

export interface CellProps {
  row: string[];
}
