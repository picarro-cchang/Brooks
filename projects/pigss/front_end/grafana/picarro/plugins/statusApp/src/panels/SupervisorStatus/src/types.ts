import { GrafanaTheme } from '@grafana/ui';

export interface SupervisorStatusProps {
}

export interface SupervisorStatusLayoutProps {
  theme: GrafanaTheme;
  options: SupervisorStatusProps;
}