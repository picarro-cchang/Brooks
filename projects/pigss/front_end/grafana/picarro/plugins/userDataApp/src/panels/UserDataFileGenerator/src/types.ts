import { GrafanaTheme } from "@grafana/ui";
import { TimeRange } from "@grafana/data";

export interface DataGeneratorPanelProps {
  timeRange: TimeRange;
}

export interface DataGeneratorLayoutProps {
  theme: GrafanaTheme;
  options: DataGeneratorPanelProps;
}
