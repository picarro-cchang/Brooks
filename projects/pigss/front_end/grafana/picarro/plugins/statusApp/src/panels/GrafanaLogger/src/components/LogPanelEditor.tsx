import React, { PureComponent } from 'react';
import { PanelOptionsGroup, PanelEditorProps, Select, FormLabel } from '@grafana/ui';
import { LogProps } from './types';
import { LEVEL_OPTIONS, LIMIT_OPTIONS } from '../constants';

const labelWidth = 6;
const selectWidth = 12;

export class LogPanelEditor extends PureComponent<PanelEditorProps<LogProps>> {
  onLevelChange = (item: any) => {
    this.props.onOptionsChange({ ...this.props.options, level: [...item] });
  };
  onLimitChange = (limit: any) => {
    this.props.onOptionsChange({ ...this.props.options, limit: limit.value });
  };

  render() {
    const { level, limit } = this.props.options;

    return (
      <PanelOptionsGroup title="Configuration">
        <div className="row">
          <div className="gf-form col-md-3 col-sm-3">
            <FormLabel width={labelWidth}>Level</FormLabel>
            <Select
              options={LEVEL_OPTIONS}
              onChange={this.onLevelChange}
              value={level}
              backspaceRemovesValue
              isMulti={true}
            />
          </div>
          <div className="gf-form col-md-3 col-sm-3">
            <FormLabel width={labelWidth}>Limit</FormLabel>
            <Select
              width={selectWidth}
              options={LIMIT_OPTIONS}
              onChange={this.onLimitChange}
              value={LIMIT_OPTIONS.find(option => option.value === limit.toString())}
              backspaceRemovesValue
            />
          </div>
        </div>
      </PanelOptionsGroup>
    );
  }
}
