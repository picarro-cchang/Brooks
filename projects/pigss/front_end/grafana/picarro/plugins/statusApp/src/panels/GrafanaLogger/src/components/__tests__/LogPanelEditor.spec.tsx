import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';
import WS from "jest-websocket-mock";
import {LogPanelEditor} from '../LogPanelEditor';
import {LogProps} from '../types';
import {PanelEditorProps} from '@grafana/ui'

const defaultProps: PanelEditorProps<LogProps> = {
    options: {
        level: [],
        limit: 1,
        timeRange: null,
        data: []
    },
    onOptionsChange: () => {},
    data: {
        state: null,
        series: null, 
        timeRange: null
    }
};

describe('<LogPanelEditor />', () => {
    const wrapper = shallow(<LogPanelEditor {...defaultProps}/>);
    it('Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});