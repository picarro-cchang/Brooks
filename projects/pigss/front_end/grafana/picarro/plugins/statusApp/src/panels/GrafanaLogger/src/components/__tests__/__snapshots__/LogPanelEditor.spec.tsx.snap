// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogPanelEditor /> Match Snapshot 1`] = `
<PanelOptionsGroup
  title="Configuration"
>
  <div
    className="row"
  >
    <div
      className="gf-form col-md-3 col-sm-3"
    >
      <FormLabel
        width={6}
      >
        Level
      </FormLabel>
      <Select
        autoFocus={false}
        backspaceRemovesValue={true}
        className=""
        components={
          Object {
            "Group": [Function],
            "IndicatorsContainer": [Function],
            "MenuList": [Function],
            "Option": [Function],
            "SingleValue": [Function],
          }
        }
        isClearable={false}
        isDisabled={false}
        isLoading={false}
        isMulti={true}
        isSearchable={true}
        maxMenuHeight={300}
        onChange={[Function]}
        openMenuOnFocus={false}
        options={
          Array [
            Object {
              "label": "10",
              "value": "10",
            },
            Object {
              "label": "20",
              "value": "20",
            },
            Object {
              "label": "30",
              "value": "30",
            },
            Object {
              "label": "40",
              "value": "40",
            },
            Object {
              "label": "50",
              "value": "50",
            },
          ]
        }
        tabSelectsValue={true}
        value={Array []}
      />
    </div>
    <div
      className="gf-form col-md-3 col-sm-3"
    >
      <FormLabel
        width={6}
      >
        Limit
      </FormLabel>
      <Select
        autoFocus={false}
        backspaceRemovesValue={true}
        className=""
        components={
          Object {
            "Group": [Function],
            "IndicatorsContainer": [Function],
            "MenuList": [Function],
            "Option": [Function],
            "SingleValue": [Function],
          }
        }
        isClearable={false}
        isDisabled={false}
        isLoading={false}
        isMulti={false}
        isSearchable={true}
        maxMenuHeight={300}
        onChange={[Function]}
        openMenuOnFocus={false}
        options={
          Array [
            Object {
              "label": "10",
              "value": "10",
            },
            Object {
              "label": "20",
              "value": "20",
            },
            Object {
              "label": "50",
              "value": "50",
            },
          ]
        }
        tabSelectsValue={true}
        width={12}
      />
    </div>
  </div>
</PanelOptionsGroup>
`;
